2996_reflection/reflection.md

```diff
@@ -4518,6 +4565,55 @@ consteval bool test_trait(info templ, R&& arguments);
 :::
 :::
 
+### [meta.reflection.result] Expression result reflection {-}
+
+::: std
+::: addu
+```cpp
+template <typename T>
+  consteval info reflect_value(T expr);
+```
+
+[#]{.pnum} *Mandates*: `T` is a structural type. `T` is not a reference type. Any subobject of the value computed by `expr` having reference or pointer type designates an entity that is a permitted result of a constant expression.
+
+[#]{.pnum} *Returns*: A reflection of the value computed by an lvalue-to-rvalue conversion applied to `expr`. The type of the reflected value is the cv-unqualified version of `T`.
+
+```cpp
+template <typename T>
+  consteval info reflect_object(T& expr);
+```
+
+[#]{.pnum} *Mandates*: `T` is not a function type. `expr` designates an entity that is a permitted result of a constant expression.
+
+[#]{.pnum} *Returns*: A reflection of the object referenced by `expr`.
+
+```cpp
+template <typename T>
+  consteval info reflect_function(T& expr);
+```
+
+[#]{.pnum} *Mandates*: `T` is a function type.
+
+[#]{.pnum} *Returns*: `^fn`, where `fn` is the function referenced by `expr`.
+
+```cpp
+template <reflection_range R = span<info const>>
+  consteval info reflect_invoke(info target, R&& args);
+template <reflection_range R1 = span<info const>, reflection_range R2 = span<info const>>
+  consteval info reflect_invoke(info target, R1&& tmpl_args, R2&& args);
+```
+
+[#]{.pnum} Let `F` be the entity reflected by `target`, let `Arg0` be the entity reflected by the first element of `args` (if any), let `Args...` be the sequence of entities reflected by the elements of `args` excluding the first, and let `TArgs...` be the sequence of entities or aliases designated by the elements of `tmpl_args`.
+
+[#]{.pnum} If `F` is a non-member function, a value of pointer to function type, or a value of closure type, then let `INVOKE-EXPR` be the expression `F(Arg0, Args...)`. Otherwise, if `F` is a member function or a value of pointer to member function type, then let `INVOKE-EXPR` be the expression `Arg0.F(Args...)` or `Arg0->F(Args...)` respectively. Otherwise, if `F` is a constructor for a class `C`, then let `INVOKE-EXPR` be the expression `C(Arg0, Args...)` for which only the constructor `F` is considered by overload resolution. Otherwise, if `F` is a non-member function template or a member function template, then let `INVOKE-EXPR` be the expression `F<TArgs...>(Arg0, Args...)` or `Arg0.template F<TArgs...>(Args...)` respectively. Otherwise, if `F` is a constructor template, then let `INVOKE-EXPR` be the expression `C(Arg0, Args...)` for which only the constructor `F` is considered by overload resolution, and `TArgs...` are inferred as explicit template arguments for `F`.
```

Comment:
For the first part, I think can lean on `INVOKE(F, Arg0, Args...)` for the first part? Cause it could also be a pointer to member data and it'd be nice to just not have to invent new things. 

I think also that avoids needing to split the arguments, since you just get `INVOKE(F, Args...)` for the first overload and `F<TArgs...>(Args...)` for the second.

Reply:
Rebased onto INVOKE.